language/stdlib/modules/vector.mvir

```diff
@@ -2,12 +2,15 @@
 
 module Vector {
 
-  native struct T;
+  // Vector containing data of type Item
+  native struct T<Element: all>;
 
-  native public length(v: &Self.T): u64;
+  // Return the length of the vector
+  native public length<Element: all>(v: &Self.T<Element>): u64;
```

Comment:
You don't need to add `: all` do you?
I thought we removed `: all` from the syntax? (at least I commented that we should) @vgao1996

Reply:
I personally think we should always require explicit constraints to avoid surprises. This generics system is so different from traditional ones that I don't think it will be obvious what the default behavior should be in the absence of constraints. Also, it helps visually distinguish generic type parameters from the type arguments to (e.g.) `move_from`.

Reply:
I disagree, the generic system is actually really similar to a normal one. `<T>` or `<T: all>` behaves just like it would in Rust for example (you cant copy or clone the value).

Reply:
We may have different taste on this; I'm just never a fan of implicit default behaviors that one has to learn.

> Also, it helps visually distinguish generic type parameters from the type arguments to (e.g.) move_from.

What are your thoughts on this?

Reply:
I don't really think it's an implicit behavior. Generics/type parameters, by default in any language, represent a place holder for any type in the language. This is exactly what `T: all` means. It seems overly redundant to me, on par with doing like `if (my_vector.is_empty() == true)`

> Also, it helps visually distinguish generic type parameters from the type arguments to (e.g.) move_from.

I don't understand this comment, as `move_from` doesn't have a signature definable in the language. 
I see the syntax of `move_from<T>` really as a pun as it is "similar" feeling to a user, but very different from a type system perspective. If we wanted to be overly pedantic I would suggest `move_from[T]` or `move_from@T` to differentiate them

Reply:
@tnowacki to answer your question, I made ": all" an optional syntax. "T" is equivalent to "T: all", but you can also write it down explicitly if you wish.

Reply:
Yeah, I see this as unnecessary. We should go all or nothing. 

As I have described above, I think we gain absolutely nothing by letting people write `T: all`

Reply:
Fair point. Let's have this discussion in a broader forum. Any blockers left on this diff?

Reply:
#699